### PR TITLE
Add: use osx-commons-subgraph

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

This PR aims to import the package `osx-commons-subgraph`. This package includes id generation functions that will be used all over the repo. 

Task ID: [OS-737](https://aragonassociation.atlassian.net/browse/OS-737)

## Type of change

See the framework lifecylce in `packages/contracts/docs/framework-lifecycle` to decide what kind of change this pull request is.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-737]: https://aragonassociation.atlassian.net/browse/OS-737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/src/packages/multisig/multisig.ts
Comment (Rekard0) [inline packages/subgraph/src/packages/multisig/multisig.ts:177]: I noticed we create the member ID for multisig and Tokenvoting using the pluginId & members's address

however i see the order of the 2 paramsare oposite to each other in each one,

can u investiage what would be the implication of unifying them on SDK, App ...etc

and if feel it should not be part this PR, please create a follow up task

File: packages/subgraph/src/packages/token/governance-erc20.ts
Comment (Rekard0) [inline packages/subgraph/src/packages/token/governance-erc20.ts:15]: related to https://github.com/aragon/osx/pull/520#discussion_r1449072166

File: packages/subgraph/src/packages/token/token-voting.ts
Comment (Rekard0) [inline packages/subgraph/src/packages/token/token-voting.ts:144]: the comment on the `TokenVotingVoter`'s ID shows that the id is an address, and not the output of `generateMemberEntityId()` 

can you verify that is everything is correct and update the schema comment

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts:70]: using `generateEntityIdFromAddress()` implies that the output will be used as an entity ID
but in reality you just want the hex string representation of this members address

also the naming of `memberOneId` seems off, it is not the addressList member (voter) id structure of (pluginId + '_' + memberAddressHexString)

the easiest suggestion here would be to use `memberOneAddress.toHexString()` and not have `memberOneId`, or you can come up with a better solution if you like.

if you agree please revisit other ocurrance of this type

@Michael-A-Heuer for visibility and second opinion

File: packages/subgraph/tests/admin/adminMembers.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/admin/adminMembers.test.ts:22]: same as https://github.com/aragon/osx/pull/520#discussion_r1449527434

File: packages/subgraph/tests/dao/dao_v1_0_0.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/dao/dao_v1_0_0.test.ts:3]: where do u use this import? 
I could not find it, can you point me to it?

File: packages/subgraph/tests/dao/dao_v1_3_0.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/dao/dao_v1_3_0.test.ts:71]: Why TODO? did you meant to use `generateTokenEntityId()`?

File: packages/subgraph/tests/helpers/method-classes.ts
Comment (Rekard0) [inline packages/subgraph/tests/helpers/method-classes.ts:547]: double check, if you don't the above case (https://github.com/aragon/osx/pull/520#discussion_r1449527434) here too

File: packages/subgraph/tests/multisig/multisig.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/multisig/multisig.test.ts:223]: I wonder if we should use the wording of "approver" here?
I know the inside the pluginEnitities we call them members,
while we are assigning `MultisigApprover` to them, in this example.

```
type MultisigPlugin implements IPlugin @entity {
  "TODO: attributes should be appended to Plugin once plugin is seperated from Aragon OSx"
  id: ID!
  dao: Dao!
  pluginAddress: Bytes!

  installations: [PluginInstallation!]! @derivedFrom(field: "plugin")

  proposalCount: BigInt
  proposals: [MultisigProposal!]! @derivedFrom(field: "plugin")
  members: [MultisigApprover!]! @derivedFrom(field: "plugin")

  # multisig settings
  minApprovals: Int
  onlyListed: Boolean
}

type MultisigApprover @entity {
  id: ID! # plugin_address + member_address
  address: String # address as string to facilitate filtering by address on the UI
  proposals: [MultisigProposalApprover!]! @derivedFrom(field: "approver")
  plugin: MultisigPlugin!
}
```

but this pattern repeats for other plugins
i think we should use "approver" in our test to reduce confutions, since the member atribute is ment for the consumers

@Michael-A-Heuer i would like to have a second opinion on this

File: packages/subgraph/src/packages/multisig/multisig.ts
Comment (josemarinas) [reply packages/subgraph/src/packages/multisig/multisig.ts:177]: Okay, in the sdk will have no implications on the app i have no idea, will ask them

File: packages/subgraph/src/packages/token/token-voting.ts
Comment (josemarinas) [reply packages/subgraph/src/packages/token/token-voting.ts:144]: true, good catch

File: packages/subgraph/tests/multisig/multisig.test.ts
Comment (josemarinas) [reply packages/subgraph/tests/multisig/multisig.test.ts:223]: I mean the tests themselves are not clear but i think its a matter of how this is structured and named

```ts
 // checks
  const memberAddress = Address.fromString(ADDRESS_ONE);

  const memberEntityId = generateMemberEntityId(pluginAddress, memberAddress);

  const voterEntityId = generateVoterEntityId(memberEntityId, proposal.id);
  // check proposalVoter
  assert.fieldEquals(
    'MultisigProposalApprover',
    voterEntityId,
    'id',
    voterEntityId
  );
  assert.fieldEquals(
    'MultisigProposalApprover',
    voterEntityId,
    'approver',
    memberEntityId
  );
  assert.fieldEquals(
    'MultisigProposalApprover',
    voterEntityId,
    'proposal',
    proposal.id
  );
  ```

If you look at this  you have `memberEntityId` and `voterEntityId`, here `voterEntityId` is the id for the `MultisigProposalApprover` so we shoyld change the name of this to `proposalApproverEntityId` or something similar, but if you go a bit down, `voterEntityId`  is the value of the field  `approver` inside the `MultisigProposalApprover` so what will be the name for this id?, `approverEntityId`?

File: packages/subgraph/tests/token/token-voting.test.ts
Comment (Rekard0) [inline packages/subgraph/tests/token/token-voting.test.ts:62]:
```suggestion
  // check TokenVotingProposal
  proposal.assertEntity();

  // check TokenVotingPlugin
```

File: packages/subgraph/tests/multisig/multisig.test.ts
Comment (Michael-A-Heuer) [reply packages/subgraph/tests/multisig/multisig.test.ts:223]: Puh, I think we should consistently name things and as you both point out, things seem to be not consistently named. 
This sounds like more work and could be worth a separate task.

File: packages/subgraph/tests/registry/daoRegistry.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/registry/daoRegistry.test.ts:50]: Use denylist/allowlist instead of blacklist/whitelist. See https://aragonassociation.atlassian.net/browse/APP-47 for the reasoning.

**Review by Michael-A-Heuer — COMMENTED** [review]
Minor comments

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts
Comment (Michael-A-Heuer) [reply packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts:70]: Yes, sounds reasonable

File: packages/subgraph/tests/registry/daoRegistry.test.ts
Comment (josemarinas) [reply packages/subgraph/tests/registry/daoRegistry.test.ts]: fixed in [7e98a90](https://github.com/aragon/osx/pull/520/commits/7e98a90c608bf5e6dfaf7a66b02d887853de5407)

File: packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts
Comment (josemarinas) [reply packages/subgraph/tests/addresslist-voting/addresslist-voting.test.ts]: renamed them in [7e98a90](https://github.com/aragon/osx/pull/520/commits/7e98a90c608bf5e6dfaf7a66b02d887853de5407)

File: packages/subgraph/src/packages/token/token-voting.ts
Comment (josemarinas) [reply packages/subgraph/src/packages/token/token-voting.ts:144]: re reviewed this and the `generateMemebrEntityId` was correct, in the past we were using the generated id, not the address id

File: packages/subgraph/src/packages/multisig/multisig.ts
Comment (josemarinas) [reply packages/subgraph/src/packages/multisig/multisig.ts:177]: Asked @sepehr2github and seem ok on their side so It seems okay

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**
